### DISCUSSION TITLE
Set culture based on domain

### DISCUSSION THREAD

**Discussion by netwavebe** [OrchardCMS/OrchardCore#13264] [Q&A]
Hi,

I have a website with content in two cultures, es-ES and nl-BE for example.

I want to set the culture solely based on the domain:

* http://www.website.es ALWAYS has 'es-ES' culture 
* http://www.website.be ALWAYS has 'nl-BE' culture

There will be no culture picker on the website, only the domain drives the culture.

How can I achieve this? I was thinking of using `builder.UseMiddleware<MyMiddleware>()` but perhaps there's a better way?

Any insight on what works well and what doesn't is greatly appreciated.

**Comment by hishamco** [comment]
If you are using a `RouteData` values you can simply use route localization, but in  your case you mght need to create a custom `RequestCultureProvider` that reads the domain extension based on that set the current culture
